### PR TITLE
feat: improve Cache-Control support

### PR DESCRIPTION
**Pull request by fiam**
- Declare cache as public by default for anonymous operations, private
for authenticated ones
- Include a must-revalidate header by default
- Allow disabling max-age and stale-while-revalidate

## Motivation and Context

By default, we were previously omitting any Cache-Control headers. This can cause problems with overeager caching by some CDN

## Marketing Changelog

Support for Cache-Control headers has been improved with better defaults, adaptability to change depending on the incoming request and more flexibility.

#### Checklist

- [x] run `make test`
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by jensneuse** [comment]
@fiam add marketing section to description maybe?

**Review by Pagebakers — CHANGES_REQUESTED** [review]
Oh this is very nice!

We should also reflect this change in the docs
https://docs.wundergraph.com/docs/wundergraph-operations-ts-reference/configure-caching

**Review by jensneuse — APPROVED** [review]

File: packages/sdk/src/configure/operations.ts [hunk 1]
```diff
@@ -13,18 +13,46 @@ export interface BaseOperationConfiguration {
...
+	/**
+	 * Mark responses as public or private. Set to undefined to omit. Notice that
+	 * any authenticated operation will override this header and set it to private.
+	 *
+	 * @default Public for operations without authentication, private otherwise
+	 */
+	public?: boolean;
+	/**
+	 * max-age component of the Cache-Control header.
+	 * Set to undefined to omit this component.
```
Comment (StarpTech) [inline packages/sdk/src/configure/operations.ts:31]: Does `0` works too?

**Review by StarpTech — APPROVED** [review]
LGTM

File: packages/sdk/src/configure/operations.ts (hunk 1 above)
Comment (fiam) [reply packages/sdk/src/configure/operations.ts:31]: It does, I've added a test for it too.

**Review by Pagebakers — APPROVED** [review]

**Pull request merged**